client/src/pages/HomePage.js

```diff
@@ -32,34 +35,42 @@ const HomePage = () => {
       "/api/mentions?platforms=reddit" +
       (searchTerm ? "&keyword=" + searchTerm : "") +
       "&page=" +
-      newPage;
+      (currentPage + 1) +
+      "&sort=" +
+      sort;
 
     const results = await axios(url, config);
     setHasMore(results.data.nextPage ? true : false);
     const newData = [...mentionDatas, results.data.mentions].flat();
     setMentionDatas(newData);
+    setCurrentPage((prev) => prev + 1);
+  };
+
+  const handleAlignment = (event, newAlignment) => {
+    setSwitching(true);
+    getMentions(dispatch, searchTerm, user.platforms, 1, newAlignment)
+      .then((data) => {
+        setMentionDatas(data.mentions);
+        setHasMore(data.nextPage ? true : false);
+      })
+      .catch((err) => alert(err.message))
+      .finally(() => setSwitching(false));
+    setSort(newAlignment);
+    setCurrentPage(1);
   };
 
   useEffect(() => {
-    getMentions(dispatch, searchTerm, user.platforms)
+    getMentions(dispatch, searchTerm, user.platforms, 1, sort)
       .then((data) => {
         setMentionDatas(data.mentions);
         setHasMore(data.nextPage ? true : false);
       })
       .catch((err) => alert("Cookie expired. Please log in again"));
+    setCurrentPage(1);
```

Comment:
nice, this was what worked for me as well but I think it could just go in the then block but doesn't matter too much!

Reply:
You're right, makes more sense to have it in the then block with the other set actions